cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -1240,18 +1258,43 @@ protected OperationHandle getOperationHandle(QueryHandle handle) throws ExploreE
     return getOperationInfo(handle).getOperationHandle();
   }
 
+  protected QueryStatus fetchStatus(OperationInfo operationInfo)
```

Comment:
`getStatus` is the public API whereas this one is internal, to be overridden by subclasses to support various versions of Hive.